src/main/java/uk/gov/hmcts/reform/pip/rules/controllers/RootController.java

```diff
@@ -62,4 +72,106 @@
+        return ResponseEntity.status(HttpStatus.OK)
+            .body(rulesService.getCourt(id));
+    }
+
+    /**
+     * This endpoint returns a court list, based on input search.
+     *
+     * <p>It will apply the business rules against the court and requesting user,
+     * to understand whether they should see the requested court.
+     *
+     * <p>If the user does not have permission to see the resource, or the resource is not found,
+     * then a 404 will be returned instead.
+     *
+     * @param input The ID of the court to search for.
+     * @return The court, if found.
+     */
+    @GetMapping("/courtlist/{input}")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, response = String.class, message = "Court has been found"),
+        @ApiResponse(code = 404, response = String.class, message = "Court has not been found")
+    })
+    public ResponseEntity<List<CourtHearings>> getCourtList(
+        @ApiParam(value = "The input search to retrieve", required = true) @PathVariable String input) {
+        return ResponseEntity.status(HttpStatus.OK)
+            .body(rulesService.getCourtList(input));
+    }
+
+    /**
+     * This endpoint returns a court list.
+     *
+     * <p>It will apply the business rules against the court and requesting user,
+     * to understand whether they should see the requested court.
+     *
+     * <p>If the user does not have permission to see the resource, or the resource is not found,
+     * then a 404 will be returned instead.
+     *
+     * @return The list of all courts, if found.
+     */
+    @GetMapping("/courtlistall")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, response = String.class, message = "Court has been found"),
+        @ApiResponse(code = 404, response = String.class, message = "Court has not been found")
+    })
+    public ResponseEntity<List<CourtHearings>> getCourtList() {
+        return ResponseEntity.status(HttpStatus.OK)
+            .body(rulesService.getCourtList(""));
+    }
+
+    /**
+     * This endpoint returns a hearings list, based on court's ID.
+     *
+     * <p>It will apply the business rules against the court and requesting user,
+     * to understand whether they should see the requested court.
+     *
+     * <p>If the user does not have permission to see the resource, or the resource is not found,
+     * then a 404 will be returned instead.
+     *
+     * @param id The ID of the court to search for.
+     * @return The hearings list, if found.
+     */
+    @GetMapping("/hearings/{id}")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, response = String.class, message = "hearings have been found"),
+        @ApiResponse(code = 404, response = String.class, message = "hearings have not been found")
+    })
+    public ResponseEntity<CourtHearings> getHearings(
+        @ApiParam(value = "The court ID to retrieve", required = true) @PathVariable Integer id) throws ParseException {
```

Comment:
youre throwing a ParseException, we should be handling errors especially if they arise at an api level

Reply:
created a new global handling class for this exception